### DISCUSSION TITLE
centering on the work grid

### DISCUSSION THREAD

**Discussion by Olddawg53** [arkypita/LaserGRBL#1987] [General]
I read you can centurion a work piece by halving the size of the piece and use a negative number. I.E. 50mm x 50mm would be -25mm x -25mm. I've yet to try this.

Right hope for me is to have the laser work from center of work grid (table) vs. front/left corner. Being new with this, I've yet to find the procedure for this setting. That is, hoping there is one.

**Comment by LsrSal** [comment]
Maybe you can describe what is the intent and why conventional laser coordinates may not be good for you?

It is a general convention for laser G code to be exclusively in positive space, vs. CNC where it is in negative.
Most laser tools, this one included, will generate positive space coordinates, as per convention.
Irrelevant of that, if you have G code generated elsewhere that has negative coordinates, once you have set work origin (like offset from home for you work piece), then this software has no problem to go wherever, including your negative coordinates. You must consider that origin/offset added to all and any coordinates for your work must fit within work space coordinates capabilities of your machine, otherwise moves beyond that maybe silently dropped and work will be messed up. Silently: not exactly, there will be warning posted over serial, so you may notice that in Terminal, but it may not stop the work, depending on your settings.

If you need to center your work around circle's center for example: once trajectory is rendered - you may click on screen YOUR center on that trajectory, laser will go there, then you may center your work piece around that.

**Reply by Olddawg53** [reply]
I like placing all projects in the center of the working field/table. Gives me better flexibility for larger projects. I find the start point of front/left to be very confining. 

As I load a project is uses the front/left start point or centers according to the project size (which is not the workspace/table size).

Im hoping to set the workspace center as the default position of any size project I do.
